src/Framework/ConversationTopic.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using StardewModdingAPI;
+using StardewValley;
+
+
+namespace QuestFramework.Framework
+{
+    public class ConversationTopicOptions
```

Comment:
Why this class is public and had internal fields? I suggest use internal class with public fields. Whole class with all public fields is always internal.

- Make his class `internal` and fialds `public`. All will be internal and available for Newtonsoft JSON for fill data with data from JSON.
- Move this class to standalone file in `QuestFramework.Framework.ContentPacks` namespace.

Reply:
> Please do it with events on `CustomQuest` class for content packs only. For programatic usage mod dev can handle these events manually and call add topic methods themself.
> 
> ```cs
> // Pseudocode in src/Framework/ContentPacks/Loader.cs
> managedQuest.Accepted += (sender, e) =>
>             {
>                 ConversationTopicMethods.AddConversationTopic(questData.ConversationTopic.AddWhenQuestAccepted);
>                 ConversationTopicMethods.RemoveConversationTopic(questData.ConversationTopic.RemoveWhenQuestAccepted);
>             };
> ```

I don't really get what does it mean by using events on CustomQuest...

Reply:
This is event listener. The Accepted field on the CustomQuest is an event. This event is called when event is accepted. And if you want to add your conversation when event is accepted for concrete quest, just handle this event, subscribe it with your event listener. That listener calls your code which will be called when accepted event is invoked. That event envoker must be called on the right place (in this case in the watchdog). Events can have more than one listener. You can imaginate your lisenners as actions which can be executed when this event raised.

More about events: https://docs.microsoft.com/en-us/dotnet/standard/events/how-to-raise-and-consume-events
Cookbook with explain: https://www.tutorialsteacher.com/csharp/csharp-event